staging/src/k8s.io/apimachinery/pkg/runtime/serializer/cbor/internal/modes/decode_test.go

```diff
@@ -45,6 +45,264 @@ func TestDecode(t *testing.T) {
 		want          interface{}
 		assertOnError func(t *testing.T, e error)
 	}{
+		{
+			name:  "reject duplicate negative int keys",
+			modes: []cbor.DecMode{modes.DecodeLax},
+			in:    hex("a220012002"), // {1:1, 1:2}
+			into:  struct{}{},
+			assertOnError: assertOnConcreteError(func(t *testing.T, e *cbor.DupMapKeyError) {
+				if diff := cmp.Diff(e, &cbor.DupMapKeyError{Key: int64(-1), Index: 1}); diff != "" {
+					t.Errorf("diff between actual error and expected error:\n %s", diff)
+				}
+			}),
+		},
+		{
+			name:  "reject duplicate positive int keys",
+			modes: []cbor.DecMode{modes.DecodeLax},
+			in:    hex("a201010102"), // {1:1, 1:2}
+			into:  struct{}{},
+			assertOnError: assertOnConcreteError(func(t *testing.T, e *cbor.DupMapKeyError) {
+				if diff := cmp.Diff(e, &cbor.DupMapKeyError{Key: int64(1), Index: 1}); diff != "" {
+					t.Errorf("diff between actual error and expected error:\n %s", diff)
+				}
+			}),
+		},
+		{
+			name: "reject duplicate text string keys into map",
+			in:   hex("A2614101614102"), // {A:1, A:2}
+			into: struct {
+				A int `json:"A"`
+			}{},
+			assertOnError: assertOnConcreteError(func(t *testing.T, e *cbor.DupMapKeyError) {
+				if diff := cmp.Diff(e, &cbor.DupMapKeyError{Key: string("A"), Index: 1}); diff != "" {
+					t.Errorf("diff between actual error and expected error:\n %s", diff)
+				}
+			}),
+		},
+		{
+			name:  "reject duplicate text string keys into struct",
+			modes: []cbor.DecMode{modes.DecodeLax},
+			in:    hex("A2614101614102"), // {A:1, A:2}
+			into:  struct{}{},
+			assertOnError: assertOnConcreteError(func(t *testing.T, e *cbor.DupMapKeyError) {
+				if diff := cmp.Diff(e, &cbor.DupMapKeyError{Key: string("A"), Index: 1}); diff != "" {
+					t.Errorf("diff between actual error and expected error:\n %s", diff)
+				}
+			}),
+		},
+		{
+			name: "reject duplicate byte string keys into map",
+			in:   hex("A2414101414102"), // {'A':1, 'A':2}
+			into: map[string]interface{}{},
+			assertOnError: assertOnConcreteError(func(t *testing.T, e *cbor.DupMapKeyError) {
+				if diff := cmp.Diff(e, &cbor.DupMapKeyError{Key: string("A"), Index: 1}); diff != "" {
+					t.Errorf("diff between actual error and expected error:\n %s", diff)
+				}
+			}),
+		},
+		{
+			name:  "reject duplicate byte string keys into struct",
+			modes: []cbor.DecMode{modes.DecodeLax},
+			in:    hex("A2414101414102"), // {'A':1, 'A':2}
+			into:  struct{}{},
+			assertOnError: assertOnConcreteError(func(t *testing.T, e *cbor.DupMapKeyError) {
+				if diff := cmp.Diff(e, &cbor.DupMapKeyError{Key: string("A"), Index: 1}); diff != "" {
+					t.Errorf("diff between actual error and expected error:\n %s", diff)
+				}
+			}),
+		},
+		{
+			name: "reject duplicate byte string & text string keys into map",
```

Comment:
Nit: & in subtest name could be awkward in an unquoted `-run` flag passed to go test.

```suggestion
			name: "reject duplicate byte string and text string keys into map",
```